packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -66,7 +66,7 @@ export async function ssrTransform(
   }
 
   function defineExport(position: number, name: string, local = name) {
-    s.appendRight(
+    s.appendLeft(
       position,
       `\nObject.defineProperty(${ssrModuleExportsKey}, "${name}", ` +
         `{ enumerable: true, configurable: true, get(){ return ${local} }});`
```

Comment:
@ygj6 This version works for me, thanks!

The only other very small thing I'd suggest is to move the `\n` from the beginning to the end of the appended string or just add another one at the end. This obviously wouldn't affect execution and is thus not necessary, but `}});function fn2() {` in the snapshot would be more legible with a newline after the `;`. Just a thought.

Either way, glad to get to the bottom of this bug!

Reply:
Hey @neb, thanks for your review and input on this PR! If you think that this can be further improved, please create a new PR, and let's discuss the addition of the new line out of the context of this bug.